src/forms/subscription-delete.ts

```diff
@@ -1,47 +1,49 @@
-import { ConfigStoreProps, KVStoreClient, KVStoreOptions } from '../clients/kvstore';
-import { AppExpandLevels, AppFieldTypes, ExceptionType, OpsGenieIcon, Routes, StoreKeys, SubscriptionCreateForm, SubscriptionDeleteForm } from '../constant';
+import { AppActingUser, AppCallRequest, AppCallValues, AppForm, AppSelectOption, Integration, Subscription, Teams } from '../types';
+import { AppExpandLevels, AppFieldTypes, ExceptionType, OpsGenieIcon, Routes, SubscriptionDeleteForm } from '../constant';
 import { OpsGenieClient, OpsGenieOptions } from '../clients/opsgenie';
 import { configureI18n } from '../utils/translations';
-import { getIntegrationsList, tryPromise } from '../utils/utils';
+import { getIntegrationsList, isUserSystemAdmin, tryPromise } from '../utils/utils';
 import { Exception } from '../utils/exception';
+import { ExtendRequired, allowMemberAction, getOpsGenieAPIKey } from '../utils/user-mapping';
+
+import { getAllTeamsCall } from './list-team';
 
 export async function subscriptionDeleteCall(call: AppCallRequest): Promise<string> {
-    const mattermostUrl: string = call.context.mattermost_site_url!;
-    const botAccessToken: string = call.context.bot_access_token!;
+    const actingUser: AppActingUser | undefined = call.context.acting_user;
+    const isSystemAdmin: boolean = isUserSystemAdmin(actingUser);
+    const allowMember: boolean = allowMemberAction(call.context);
     const values: AppCallValues | undefined = call.values;
     const i18nObj = configureI18n(call.context);
+    const apiKey = getOpsGenieAPIKey(call);
 
+    const optionsOps: OpsGenieOptions = {
+        api_key: apiKey,
+    };
+    const opsGenieClient: OpsGenieClient = new OpsGenieClient(optionsOps);
     const subscription: AppSelectOption = values?.[SubscriptionDeleteForm.SUBSCRIPTION_ID];
 
-    const options: KVStoreOptions = {
-        mattermostUrl,
-        accessToken: botAccessToken,
-    };
-    const kvStore: KVStoreClient = new KVStoreClient(options);
+    if (!allowMember) {
+        throw new Exception(ExceptionType.MARKDOWN, i18nObj.__('general.validation-user.genie-action-invalid'));
+    }
 
-    const configStore: ConfigStoreProps = await kvStore.kvGet(StoreKeys.config);
+    if (!isSystemAdmin) {
+        const opsSubscription: Integration = await tryPromise<Integration>(opsGenieClient.getIntegration(subscription.value), ExceptionType.MARKDOWN, i18nObj.__('forms.error'));
 
-    const optionsOps: OpsGenieOptions = {
-        api_key: configStore.opsgenie_apikey,
-    };
-    const opsGenieClient: OpsGenieClient = new OpsGenieClient(optionsOps);
+        const teams: Teams[] = await getAllTeamsCall(call);
+        const teamsIds: string[] = teams.map((team) => team.id);
+
+        if (!teamsIds.includes(opsSubscription.ownerTeam.id)) {
+            throw new Exception(ExceptionType.MARKDOWN, i18nObj.__('binding.binding.command-delete-no-found'));
+        }
+    }
```

Comment:
Required: Are we checking the acting user's membership of the teams here?

Reply:
No, only regular members. System admin users are not email validated